src/clients/near-kit.ts

```diff
@@ -155,81 +153,80 @@ interface BalanceResults {
 }
 
 /**
- * NEAR blockchain implementation of the bridge client.
- * Handles token deployment, binding, and transfer operations on the NEAR blockchain.
+ * NEAR blockchain implementation of the bridge client using near-kit.
+ * Unified client that works with both server-side (privateKey) and browser (wallet) scenarios.
  */
 export class NearBridgeClient {
   public bitcoinService: BitcoinService
   public zcashService?: ZcashService
   private readonly utxoServices: Partial<Record<UtxoChain, UtxoChainService>> = {}
+  private readonly near: Near
+  private readonly lockerAddress: string
+  private readonly defaultSignerId: string | undefined
 
   /**
    * Creates a new NEAR bridge client instance
-   * @param wallet - NEAR account instance for transaction signing
+   * @param near - Near instance from near-kit (configured with network and credentials)
    * @param lockerAddress - Address of the token locker contract
-   * @throws {Error} If locker address is not configured
+   * @param options - Optional configuration (e.g., zcashApiKey, defaultSignerId)
    */
   constructor(
-    private wallet: Account,
-    private lockerAddress: string = addresses.near.contract,
-    options?: { zcashApiKey?: string } | string,
+    near: Near,
+    lockerAddress: string = addresses.near.contract,
+    private readonly options: { zcashApiKey?: string; defaultSignerId?: string } = {},
   ) {
-    if (lockerAddress) {
-      this.lockerAddress = lockerAddress
-    }
-    // Configure BigInt serialization for JSON.stringify
-    // biome-ignore lint/suspicious/noExplicitAny: TS will complain that `toJSON()` does not exist on BigInt
-    // biome-ignore lint/complexity/useLiteralKeys: TS will complain that `toJSON()` does not exist on BigInt
-    ;(BigInt.prototype as any)["toJSON"] = function () {
-      // The contract can't accept `origin_nonce` as a string, so we have to serialize it as a number.
-      // However, this can cause precision loss if the number is too large. We'll check if it's safe to convert
-      // and if not, we'll serialize it as a string and the contract will have to handle it.
-      const maxSafe = BigInt(Number.MAX_SAFE_INTEGER)
-      if (this <= maxSafe) {
-        return Number(this)
-      }
-      return this.toString()
-    }
+    this.near = near
+    this.lockerAddress = lockerAddress
+    this.defaultSignerId = options.defaultSignerId
 
     // Initialize Bitcoin service
     this.bitcoinService = new BitcoinService(addresses.btc.apiUrl, addresses.btc.network, {
```

Comment:
Are we sure we don't need an override anymore? I believe it was added before `MaxGasFee` was a thing so is there something else we're missing?

Reply:
Wrong place referenced, talking about the BigInt toJSON override